File: src/main/java/com/github/unchama/util/collection/ImmutableListFactory.kt
package com.github.unchama.util.collection


object ImmutableListFactory {

  @JvmStatic
  // Checked, no kotlin usage found
  fun <E> of(): List<E> {
    return emptyList()
  }

  // Checked, no kotlin usage found
  @JvmStatic
  fun <E> of(o: E): List<E> {
    return listOf(o)
  }

  @SafeVarargs
  @JvmStatic
  // Checked, no kotlin usage found
  fun <E> of(vararg o: E): List<E> {
    return listOf(*o)
  }
}

File: src/main/java/com/github/unchama/util/collection/ImmutableListFactory.scala
package com.github.unchama.util.collection

import scala.jdk.CollectionConverters._

object ImmutableListFactory {

  // Checked, no scala usage found
  def of[E](): java.util.List[E] = Nil.asJava

  // Checked, no scala usage found
  def of[E](o: E): java.util.List[E] = List(o).asJava

  def of[E](o: Array[E]): java.util.List[E] = o.toList.asJava

  @SafeVarargs
  // Checked, no scala usage found
  def of[E](o: E*): java.util.List[E] = List(o: _*).asJava

}
